mono/metadata/object-internals.h

```diff
@@ -1477,7 +1477,11 @@ typedef struct {
 static inline MonoInternalThread*
 mono_internal_thread_handle_ptr (MonoInternalThreadHandle h)
 {
-	return MONO_HANDLE_RAW (h); /* Safe */
+	/* The SUPPRESS here prevents a Centrinel warning due to merely seeing this
```

Comment:
Bonus points for macros. :) (seriously, no, macros are awful)

Reply:
I like the function, but why? You want type-specific search for MONO_HANDLE_RAW? Or abstract away the macros? Or hurts to press or read SHIFT KEY? (yes to both for me, macros are evil due to lack of scoping and inability to portabily evaluate parameters just once, and difficult writing expressions, gcc fixes some of that, but the all caps, while a useful conventional hint, is also ugly).

Reply:
In the back of my mind I've always had a feeling that if we end up with too many uses of `MONO_HANDLE_RAW`, it will become difficult to audit them all.  Hence it's better to have a use-specific version.  The "always-pinned" types like MonoInternalThread are a good candidate for a special function, IMO, because if we're going to audit all uses of `_RAW`, it'll be nice to just skip over all of them at once.

Anyway, this PR is just to suppress a Centrinel warning from every single time that `object-internals.h` is included, which should make the Centrinel report less noisy.